optimized/state_calc.py

```
@@ -0,0 +1,351 @@
+        :param npt.NDArray current_state: A 2D array where each row represents the current state of the cyclins.
+        :returns: npt.NDArray: The next state of the cyclins.
+        """
+        # batch_size, n_cyclins = current_state.shape
+
+        # state_values = jnp.matmul(current_state)
+
+        # Start with a copy of the current state
+        next_state = current_state.copy()
+
+        # Get the row for the specific cyclin (incoming edges to this cyclin)
+        cyclin_row = graph_matrix[cyclin_ix]
```

Ruff (0.12.2)

Undefined name `graph_matrix`

(F821)